crates/aptos-metrics-core/src/avg_counter.rs

```diff
@@ -0,0 +1,65 @@
+
+// Copyright © Aptos Foundation
+
+use prometheus::{
+    register_counter, register_int_counter, Counter, IntCounter,
+};
+
+
+pub struct AverageCounter {
+    sum: Counter,
+    count: IntCounter,
+}
+
+impl AverageCounter {
+    pub fn register(name: &str, desc: &str) -> AverageCounter {
+        AverageCounter {
+            sum: register_counter!(
+                format!("{}_sum", name),
+                format!("{}. Sum part of the counter", desc),
+            )
+            .unwrap(),
+            count: register_int_counter!(
+                format!("{}_count", name),
+                format!("{}. Count part of the counter", desc),
+            )
+            .unwrap(),
+        }
+    }
+
+    pub fn observe(&self, value: f64) {
+        if value != 0.0 {
```

Comment:
Why is this check needed? Does the counter not support incrementing by 0? What happens if this is negative?

Reply:
oh it works, microoptimization only :)

I can remove the check if you don't like it :)